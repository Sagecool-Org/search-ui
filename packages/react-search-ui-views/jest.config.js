module.exports = {
  testEnvironment: "jest-environment-jsdom",
  preset: "ts-jest/presets/js-with-ts-esm",
  testPathIgnorePatterns: ["./lib"],
  setupFilesAfterEnv: ["<rootDir>/src/setupTests.js"],
  moduleNameMapper: {
    "\\.(css)$": "<rootDir>/__mocks__/styleMock.js"
  }
};
