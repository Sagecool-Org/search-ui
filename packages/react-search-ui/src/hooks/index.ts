export { useSearch } from "./use_search";
