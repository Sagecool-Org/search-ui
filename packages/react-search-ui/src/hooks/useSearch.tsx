import { SearchContextState } from "@elastic/search-ui";
import { useContext, useEffect, useState } from "react";

import SearchContext from "../SearchContext";

/**
 * React hook that provides access to Search UI state and actions
 *
 * @param mapContextToProps Optional function to select specific parts of the context
 * @returns Selected search context state and actions
 */
export function useSearch<T = SearchContextState>(
  mapContextToProps?: (context: SearchContextState) => T
): T {
  const context = useContext(SearchContext);

  if (!context) {
    throw new Error("useSearch must be used within a SearchProvider");
  }

  const [state, setState] = useState<T>(() =>
    mapContextToProps
      ? mapContextToProps({
          ...context.driver.getState(),
          ...context.driver.getActions()
        })
      : ({
          ...context.driver.getState(),
          ...context.driver.getActions()
        } as T)
  );

  useEffect(() => {
    const subscription = (newState: Partial<SearchContextState>) => {
      setState((prevState: T) => {
        const fullContext = {
          ...(prevState as any),
          ...newState
        };
        return mapContextToProps
          ? mapContextToProps(fullContext)
          : (fullContext as T);
      });
    };

    context.driver.subscribeToStateChanges(subscription);
    return () => {
      context.driver.unsubscribeToStateChanges(subscription);
    };
  }, [context.driver, mapContextToProps]);

  return state;
}

export default useSearch;
