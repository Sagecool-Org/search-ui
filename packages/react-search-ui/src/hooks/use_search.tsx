import { createContext, useContext } from "react";

import type { SearchContextState } from "@elastic/search-ui";

export const UseSearchContext = createContext<SearchContextState | undefined>(
  undefined
);

export const useSearch = () => {
  const context = useContext(UseSearchContext);
  if (context === undefined) {
    throw new Error("useSearch must be used within a SearchContextProvider");
  }
  return context;
};

export default useSearch;
