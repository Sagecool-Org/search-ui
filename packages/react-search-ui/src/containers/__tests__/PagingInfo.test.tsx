import React from "react";
import { shallow } from "enzyme";
import PagingInfoContainer from "../PagingInfo";
import { PagingInfoViewProps } from "@elastic/react-search-ui-views";
import { useSearch } from "../../hooks";

jest.mock("../../hooks", () => ({
  useSearch: jest.fn()
}));

const params = {
  pagingStart: 1,
  pagingEnd: 20,
  resultSearchTerm: "test",
  totalResults: 100
};

beforeEach(() => {
  jest.clearAllMocks();

  (useSearch as jest.Mock).mockReturnValue(params);
});

it("supports a render prop", () => {
  const render = ({ start, end }: PagingInfoViewProps) => {
    return (
      <div>
        {start}
        {end}
      </div>
    );
  };
  const wrapper = shallow(<PagingInfoContainer view={render} />).dive();
  expect(wrapper).toMatchSnapshot();
});

it("renders when it doesn't have any results or a result search term", () => {
  (useSearch as jest.Mock).mockReturnValue({
    ...params,
    resultSearchTerm: ""
  });
  const wrapper = shallow(<PagingInfoContainer />).dive();
  expect(wrapper).toMatchSnapshot();
});

it("passes className through to the view", () => {
  let viewProps: PagingInfoViewProps;
  const className = "test-class";
  shallow(
    <PagingInfoContainer
      className={className}
      view={(props) => {
        viewProps = props;
        return <div />;
      }}
    />
  ).dive();
  expect(viewProps.className).toEqual(className);
});

it("passes data-foo through to the view", () => {
  let viewProps;
  const data = "bar";
  shallow(
    <PagingInfoContainer
      data-foo={data}
      view={(props) => {
        viewProps = props;
        return <div />;
      }}
    />
  ).dive();
  expect(viewProps["data-foo"]).toEqual(data);
});
