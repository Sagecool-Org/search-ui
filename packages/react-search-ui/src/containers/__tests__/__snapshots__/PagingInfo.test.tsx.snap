// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when it doesn't have any results or a result search term 1`] = `
<div>
  <div
    class="sui-paging-info"
  >
    Showing
     
    <strong>
      1
       - 
      20
    </strong>
     
    out of 
    <strong>
      100
    </strong>
  </div>
</div>
`;

exports[`supports a render prop 1`] = `
<div>
  <div>
    1
    20
  </div>
</div>
`;
